pom.xml

```diff
@@ -284,9 +302,14 @@
     </dependency>
     <dependency>
       <groupId>org.eclipse.jetty.websocket</groupId>
-      <artifactId>websocket-server</artifactId> <!-- or javax-websocket-server-impl -->
+      <artifactId>websocket-jetty-server</artifactId> <!-- or websocket-javax-server -->
       <version>${jetty.version}</version>
     </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-jdk14</artifactId>
+      <version>2.0.0-alpha5</version>
+    </dependency>
```

Comment:
Not sure I see the reason for adding this dependency. Seems to work fine without this change for me.

Reply:
do you have logs without this?

Reply:
because Jetty 10 is based only on slf4j so need a slf4j implementation available in the classpath

Reply:
Good point. Without this I get:

```
SLF4J: No SLF4J providers were found.
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#noProviders for further details.
```

So I guess this is necessary.

Reply:
but version 2?  why is jetty using an [unstable / experimental](https://www.slf4j.org/download.html#content) version.

will this impact jenkins and all plugins?  Many plugins are using slf4j APIs - do we have a another potential guavopolitc issue here?

Reply:
> but version 2? why is jetty using an [unstable / experimental](https://www.slf4j.org/download.html#content) version.
> 
> > SLF4J API version 2.0.0 relies on the [ServiceLoader](https://docs.oracle.com/javase/8/docs/api/java/util/ServiceLoader.html) mechanism to find its logging backend. See the relevant [FAQ entry](https://www.slf4j.org/faq.html#changesInVersion200) for more details.
> 
> does this impact jenkins and all plugins too?

Jetty switched from it's own rather crusty and bug ridden logging facade to slf4j.
Version 2.0.0 is the first version to support JPMS, which Jetty 10 and Jetty 11 also support.
Our use of slf4j 2.0.0 is light weight enough that you can use slf4j 1.7.x as a replacement (we don't use the fluent logging API in slf4j 2.0.0).
This is what the Eclipse RT (and IDE) folks are doing.
Just know that if you come into it this way (by using slf4j 1.7.x instead of 2.0.0), you have a broken JPMS module system.

Reply:
Also note that dropwizard installations using Jetty 10 or Jetty 11 are sticking with slf4j 2.0.0 and have had no complaints.

The label "unstable / experimental" with slf4j 2.0.0 refers to the new APIs (which Jetty doesn't use) and internal impl lookups (only Jetty's optional `jetty-slf4j-impl` uses this) as those are still evolving.
The actual implementations surrounding 2.0.0 are quite stable at runtime, many millions of Jetty 10 and Jetty 11 installations use exactly this setup. (common setups are sfl4j2 to logback, or log4j2, or java.util.logging impls)

Reply:
I would not be scare about the version naming of slf4j.
alpha5 has been here since august 2021 and alpha6 has very minor changes.
It's just using a more conservative naming approach/release strategy and not release every changes every day.

Reply:
Anyway not sure we really need jpms friendly for Jenkins. So slf4j older versions is an option. And to align with the version range used by Jenkins core. (And being a bit conservative)
I don't really have strong opinions here.